napari/_vispy/overlays/scale_bar.py

```diff
@@ -171,34 +188,57 @@ def _on_zoom_change(self, *, force: bool = False):
         )
 
         # Update scalebar and text
-        self.node.transform.scale = [sign * scale, 1, 1, 1]
+        self.line_node.transform.scale = [sign * scale, 1, 1, 1]
         self.text_node.text = f'{new_dim:~}'
 
     def _on_data_change(self):
-        """Change color and data of scale bar."""
-        if self._viewer.scale_bar.colored:
+        """Change color and data of scale bar and box."""
+
+        if self._viewer.scale_bar.color is None:
             color = self._default_color
         else:
-            # the reason for using the `as_hex` here is to avoid
-            # `UserWarning` which is emitted when RGB values are above 1
-            background_color = get_theme(
-                self._viewer.theme, False
-            ).canvas.as_hex()
-            background_color = transform_color(background_color)[0]
-            color = np.subtract(1, background_color)
-            color[-1] = background_color[-1]
+            color = self._viewer.scale_bar.color
+
+        if self._viewer.scale_bar.box_color is None:
+            box_color = self._default_box_color
+        else:
+            box_color = transform_color(self._viewer.scale_bar.box_color)
+
+        # transfor_color returns a 2d array !?
+        box_color = box_color.ravel()
```

Comment:
```suggestion
            box_color = self._viewer.scale_bar.box_color
```

I think you should be able to remove most of this since the `box_color` validator calls `transform_color` and returns the first element, also negating the need for `ravel`.